src/main/java/io/github/jumperonjava/customcursor/CursorEditScreen.java

```diff
@@ -38,137 +34,196 @@ public void close() {
-        addDrawableChild(size);
-        addDrawableChild(field);
+        var imagePathField = new TextFieldWidget(client.textRenderer, columns[0], rows[4], twoColumnSizePadding, rowSizePadding, Text.empty());
+        var sizeSlider = new SliderWidget(columns[0], rows[1], twoColumnSizePadding, rowSizePadding, Text.translatable("customcursor.edit.size"), 0, maxsize, this.targetConfig.y, 1);
+        var xPosSlider = new SliderWidget(columns[0], rows[2], columnSizePadding, rowSizePadding, Text.translatable("customcursor.edit.x"), 0., 1., this.targetConfig.x, 1 / 64f);
+        var yPosSlider = new SliderWidget(columns[1], rows[2], columnSizePadding, rowSizePadding, Text.translatable("customcursor.edit.y"), 0., 1., this.targetConfig.y, 1 / 64f);
+
+
+        sizeSlider.setValueOwn(this.targetConfig.size / maxsize);
+
+        xPosSlider.setChangedListener(d -> this.targetConfig.x = (float) (double) d);
+        yPosSlider.setChangedListener(d -> this.targetConfig.y = (float) (double) d);
+        sizeSlider.setChangedListener(d -> this.targetConfig.size = (int) (double) d);
+
+        imagePathField.setMaxLength(512);
+        imagePathField.setText(this.targetConfig.identifier.toString());
+        imagePathField.setChangedListener((s) -> {
+            try {
+                setIdentifier(Identifier.tryParse(s));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+
+        addDrawableChild(toggleWidget.build());
+        addDrawableChild(confrmButton.build());
+        addDrawableChild(cancelButton.build());
+        addDrawableChild(folderButton.build());
+        addDrawableChild(imagePathField);
+        addDrawableChild(sizeSlider);
+        addDrawableChild(xPosSlider);
+        addDrawableChild(yPosSlider);
+        addDrawable(this::renderCheckerboard);
+        addDrawable(this::renderPreview);
     }
 
     private void setIdentifier(Identifier identifier) {
         this.targetConfig.identifier = identifier;
     }
 
     private void confirm(ButtonWidget buttonWidget) {
-        try{
+        try {
             onSuccess.accept(targetConfig);
+        } catch (Exception e) {
+            e.printStackTrace();
         }
-        catch (Exception e){e.printStackTrace();}
         this.close();
     }
-    private float bgx=0,bgy=0,color,rot;
+
+
+    //? if <= 1.20.1 {
     @Override
     public void render(DrawContext context, int mouseX, int mouseY, float delta) {
-        int heightOffset = (int) (24*2.75   );
-        //? if <= 1.20.1 {
         renderBackground(context);
-        //?} else {
-        /*renderBackground(context,mouseX,mouseY,delta);
-        *///?}
         super.render(context, mouseX, mouseY, delta);
-        var vec = new Vec2f(mouseX-width/2,mouseY-height/2).normalize().multiply(delta);
-        bgx += vec.x;
-        bgy += vec.y;
-        color+=delta*0.05;
-        var n = 2;
-        renderCheckerboard(
+    }
+    //?}
```

Comment:
overriden render method with background render needed only in version 1.20.1 and earlier, all versions after this have renderBackground method called in original render method

Render of everything else is moved to init method with addDrawable()